Exercise_3.py

```diff
@@ -1,20 +1,59 @@
-# Node class  
+"""
+
+  Student : Shahreeen Shahjahan Psyche
+
+  Memory Complexity: O(N)
+
+  The Code Ran Successfully
+
+"""
+# Node class
 class Node:  
   
     # Function to initialise the node object  
-    def __init__(self, data):  
+    def __init__(self, data):
+        self.data = data
+        self.next = None
         
 class LinkedList: 
   
-    def __init__(self): 
-        
-  
-    def push(self, new_data): 
-        
+    def __init__(self):
+        self.head = None
+
+
+    # Running Complexity : O(N)
+    def push(self, new_data):
+        new_node = Node(new_data)
+        if self.head == None:
+            self.head = new_node
+        else:
+            current = self.head
+            while(current is not None):
+                if current.next is None:
+                    current.next = new_node
+                    break
+                current = current.next
   
     # Function to get the middle of  
-    # the linked list 
-    def printMiddle(self): 
+    # the linked list
+
+    # Running Complexity : O(N)
+    def printMiddle(self):
+
+        if self.head == None:
+            return
+
+        if self.head.next is None:
+            print(self.head.data)
+            return
+        fast = self.head
+        slow = self.head
+
+        while(fast.next is not None):
```

Comment:
This solution works for odd number of elements, If there's another element pushed, the code breaks. 
Check how you could fix it